rust/sdk-processor/src/processors/events_processor.rs

```diff
@@ -0,0 +1,102 @@
+use crate::steps::events_processor::{EventsExtractor, EventsStorer};
+use crate::{
+    config::indexer_processor_config::IndexerProcessorConfig,
+    steps::common::latest_processed_version_tracker::LatestVersionProcessedTracker,
+    utils::{
+        database::{new_db_pool, run_migrations, ArcDbPool},
+        starting_version::get_starting_version,
+    },
+};
+use anyhow::Result;
+use aptos_indexer_processor_sdk::{
+    aptos_indexer_transaction_stream::TransactionStreamConfig,
+    builder::ProcessorBuilder,
+    instrumented_channel::instrumented_bounded_channel,
+    steps::{TimedBuffer, TransactionStreamStep},
+    traits::{IntoRunnableStep, RunnableStepWithInputReceiver},
+};
+use std::time::Duration;
+
+pub struct EventsProcessor {
+    pub config: IndexerProcessorConfig,
+    pub db_pool: ArcDbPool,
+}
+
+impl EventsProcessor {
+    pub async fn new(config: IndexerProcessorConfig) -> Result<Self> {
+        let conn_pool = new_db_pool(
+            &config.db_config.postgres_connection_string,
+            Some(config.db_config.db_pool_size),
+        )
+        .await
+        .expect("Failed to create connection pool");
+
+        Ok(Self {
+            config,
+            db_pool: conn_pool,
+        })
+    }
+
+    pub async fn run_processor(self) -> Result<()> {
+        // (Optional) Run migrations
+        run_migrations(
```

Comment:
IMO schema should be managed outside of processor.

Reply:
Do you have a proposal on how deployments would work in that case?

Reply:
Just have another binary/script/github workflow/etc.. Schema changes are in general considered very risky, they should be explicitly acknowledged and carefully monitored, rather than shipping with binary implicitly.

Reply:
Good points, I'll create a future task to track this since it will require more testing, deployment changes and communication to partners who are running the processors.

Let's limit the scope of this PR as only for migrating existing processor to SDK.